CMakeLists.txt

```diff
@@ -34,13 +34,6 @@ include(cmake/Config.cmake)
 # https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html
 include(GNUInstallDirs)
 
-# add an option for choosing the build type (shared or static)
-if(NOT SFML_OS_ANDROID)
-    sfml_set_option(BUILD_SHARED_LIBS TRUE BOOL "TRUE to build SFML as shared libraries, FALSE to build it as static libraries")
-else()
-    set(BUILD_SHARED_LIBS TRUE)
```

Comment:
I decided to remove this setting for Android. It seems heavy-handed to forbid Android users from using static libraries.

Reply:
Android programs are java programs. Native code always come from a dynamic library.

Reply:
I don't think we _need_ to do any of that. Those on systems with particular requires like this can easily modify `BUILD_SHARED_LIBS` or `CMAKE_POSITION_INDEPENDENT_CODE` as needed. I will acknowledge however that removing this is a bit outside the scope of this PR so I'll roll it back if it risks stalling the PR.

Reply:
Right, configuring the android build might belong to android specific build scripts (maybe build.gradle or Android.mk, not sure). I suggest we test the android example before breaking stuff.

Reply:
Fair. I don't personally want to test those Android build scripts nor wait on someone else to do that so I'll remove that change for the sake of this PR being more mergeable.